ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/attestation/utils/RewardBasedAttestationSorterTest.java

```diff
@@ -0,0 +1,365 @@
+            STATE_SLOT.minus(1),
+            CURRENT_EPOCH,
+            List.of(0),
+            List.of(TIMELY_SOURCE_FLAG_INDEX, TIMELY_TARGET_FLAG_INDEX));
+
+    // Att2 (Val 0): Timely Source.
+    // Initial Reward (if val 0 has no flags): 1000*14 = 14000
+    // After Att1, val 0 has TS flag, so reward for TS becomes 0. Reward for Att2 -> 0.
+    final PooledAttestationWithData att2 =
+        mockPooledAttestationWithData(
+            STATE_SLOT.minus(2), CURRENT_EPOCH, List.of(0), List.of(TIMELY_SOURCE_FLAG_INDEX));
+
+    // Att3 (Val 1): Timely Target. Initial Reward: 1000*26 = 26000
+    // Unaffected by Att1's participation changes for Val 0.
+    final PooledAttestationWithData att3 =
+        mockPooledAttestationWithData(
+            STATE_SLOT.minus(3), CURRENT_EPOCH, List.of(1), List.of(TIMELY_TARGET_FLAG_INDEX));
+
+    // Att4 (Val 2, Previous Epoch): Timely Head. Reward: 1000*14 = 14000. Should not be recomputed.
+    final PooledAttestationWithData att4 =
+        mockPooledAttestationWithData(
+            STATE_SLOT.minus(35), // Slot in previous epoch
+            PREVIOUS_EPOCH,
+            List.of(2),
+            List.of(TIMELY_HEAD_FLAG_INDEX));
+
+    final List<PooledAttestationWithData> attestations = List.of(att1, att2, att3, att4);
+
+    // Initial rewards computed:
+    // Att1: 40000 (Val 0: TS+TT)
+    // Att2: 14000 (Val 0: TS)
+    // Att3: 26000 (Val 1: TT)
+    // Att4: 14000 (Val 2: TH, Previous Epoch)
+    // Initial Queue Order (highest reward first): Att1, Att3, Att2 (or Att4), Att4 (or Att2)
+
+    final List<RewardBasedAttestationSorter.PooledAttestationWithRewardInfo> sortedAttestations =
+        sorter.sort(attestations, 4);
+
+    // Step 1: Att1 (40000) is chosen.
+    // Validator 0 participation updated with TS_FLAG | TT_FLAG.
+    // Att2 and Att3 (current epoch) are re-evaluated. Att4 (previous epoch) is not.
+    //  - Re-evaluating Att2 (Val 0, wants TS): Val 0 now has TS_FLAG. Reward contribution for TS is
+    // 0.
+    //    New Att2 reward = 0.
+    //  - Re-evaluating Att3 (Val 1, wants TT): Unchanged as it's for Val 1. Reward = 26000.
+    // New Queue Order (Att1 already picked): Att3 (26000), Att4 (14000), Att2 (0)
+    // (Note: if Att2 and Att4 had same initial reward, their relative order might depend on
+    // original list order or other factors before re-evaluation)
+
+    // Step 2: Att3 (26000) is chosen.
+    // Validator 1 participation updated with TT_FLAG.
+    // Att2 (current epoch) is re-evaluated. Att4 (previous epoch) is not.
+    //  - Re-evaluating Att2 (Val 0, wants TS): Still 0 reward.
+    // New Queue Order (Att1, Att3 picked): Att4 (14000), Att2 (0)
+
+    // Step 3: Att4 (14000) is chosen.
+    // Validator 2 (previous epoch) participation updated.
+    // Att2 (current epoch) is NOT re-evaluated because Att4 is previous epoch.
+    // New Queue Order (Att1, Att3, Att4 picked): Att2 (0)
+
+    // Step 4: Att2 (0) is chosen.
```

Comment:
should probably be in a long `/* */` block so that maybe it collapses by default... maybe intellij collapses these too, not sure...